Two tests are failing in 5.2.1 with Python 3.13.0a5

With the latest alpha release of Python 3.13 two tests are failing again. This happens in both version 5.2.0 and version 5.2.1 with Python 3.13.0a5.

In issue #1295 @sanjayankur31 reported all tests succeeding in version 5.2.0. But that was with Python 3.13.0a4.

```python
=================================== FAILURES ===================================
_____________________ TestNibabelDeprecator.test_dep_func ______________________

self = <nibabel.tests.test_deprecated.TestNibabelDeprecator object at 0x7f7f64c8c980>

    def test_dep_func(self):
        # Test function deprecation
        dec = self.dep_func
        func = dec('foo')(func_no_doc)
        with pytest.deprecated_call():
            assert func() is None
        assert func.__doc__ == 'foo\n'
        func = dec('foo')(func_doc)
        with pytest.deprecated_call() as w:
            assert func(1) is None
            assert len(w) == 1
        assert func.__doc__ == 'A docstring\n\nfoo\n'
        func = dec('foo')(func_doc_long)
        with pytest.deprecated_call() as w:
            assert func(1, 2) is None
            assert len(w) == 1
>       assert (
            func.__doc__
            == f'A docstring\n   \n   foo\n   \n{indent(TESTSETUP, "   ", lambda x: True)}'
            f'   Some text\n{indent(TESTCLEANUP, "   ", lambda x: True)}'
        )
E       assert 'A docstring\n\nfoo\n\n\n\n.. testsetup::\n\n    >>> import pytest\n    >>> import warnings\n    >>> _suppress_warnings = pytest.deprecated_call()\n    >>> _ = _suppress_warnings.__enter__()\n\nSome text\n\n\n.. testcleanup::\n\n    >>> warnings.warn("Avoid error if no doctests to run...", DeprecationWarning)\n    >>> _ = _suppress_warnings.__exit__(None, None, None)\n\n' == 'A docstring\n   \n   foo\n   \n   \n   \n   .. testsetup::\n   \n       >>> import pytest\n       >>> import warnings\n       >>> _suppress_warnings = pytest.deprecated_call()\n       >>> _ = _suppress_warnings.__enter__()\n   \n   Some text\n   \n   \n   .. testcleanup::\n   \n       >>> warnings.warn("Avoid error if no doctests to run...", DeprecationWarning)\n       >>> _ = _suppress_warnings.__exit__(None, None, None)\n   \n'
E           A docstring
E         -    
E         -    foo
E         -    
E         -    
E         -    
E         + 
E         + foo
E         + 
E         + 
E         + 
E         -    .. testsetup::
E         ? ---
E         + .. testsetup::
E         -    
E         + 
E         -        >>> import pytest
E         ? ---
E         +     >>> import pytest
E         -        >>> import warnings
E         ? ---
E         +     >>> import warnings
E         -        >>> _suppress_warnings = pytest.deprecated_call()
E         ? ---
E         +     >>> _suppress_warnings = pytest.deprecated_call()
E         -        >>> _ = _suppress_warnings.__enter__()
E         ? ---
E         +     >>> _ = _suppress_warnings.__enter__()
E         -    
E         + 
E         -    Some text
E         ? ---
E         + Some text
E         -    
E         -    
E         + 
E         + 
E         -    .. testcleanup::
E         ? ---
E         + .. testcleanup::
E         -    
E         + 
E         -        >>> warnings.warn("Avoid error if no doctests to run...", DeprecationWarning)
E         ? ---
E         +     >>> warnings.warn("Avoid error if no doctests to run...", DeprecationWarning)
E         -        >>> _ = _suppress_warnings.__exit__(None, None, None)
E         ? ---
E         +     >>> _ = _suppress_warnings.__exit__(None, None, None)
E         -    
E         +

nibabel/tests/test_deprecator.py:95: AssertionError
_______________________ TestDeprecatorFunc.test_dep_func _______________________

self = <nibabel.tests.test_deprecator.TestDeprecatorFunc object at 0x7f7f64c8f7a0>

    def test_dep_func(self):
        # Test function deprecation
        dec = self.dep_func
        func = dec('foo')(func_no_doc)
        with pytest.deprecated_call():
            assert func() is None
        assert func.__doc__ == 'foo\n'
        func = dec('foo')(func_doc)
        with pytest.deprecated_call() as w:
            assert func(1) is None
            assert len(w) == 1
        assert func.__doc__ == 'A docstring\n\nfoo\n'
        func = dec('foo')(func_doc_long)
        with pytest.deprecated_call() as w:
            assert func(1, 2) is None
            assert len(w) == 1
>       assert (
            func.__doc__
            == f'A docstring\n   \n   foo\n   \n{indent(TESTSETUP, "   ", lambda x: True)}'
            f'   Some text\n{indent(TESTCLEANUP, "   ", lambda x: True)}'
        )
E       assert 'A docstring\n\nfoo\n\n\n\n.. testsetup::\n\n    >>> import pytest\n    >>> import warnings\n    >>> _suppress_warnings = pytest.deprecated_call()\n    >>> _ = _suppress_warnings.__enter__()\n\nSome text\n\n\n.. testcleanup::\n\n    >>> warnings.warn("Avoid error if no doctests to run...", DeprecationWarning)\n    >>> _ = _suppress_warnings.__exit__(None, None, None)\n\n' == 'A docstring\n   \n   foo\n   \n   \n   \n   .. testsetup::\n   \n       >>> import pytest\n       >>> import warnings\n       >>> _suppress_warnings = pytest.deprecated_call()\n       >>> _ = _suppress_warnings.__enter__()\n   \n   Some text\n   \n   \n   .. testcleanup::\n   \n       >>> warnings.warn("Avoid error if no doctests to run...", DeprecationWarning)\n       >>> _ = _suppress_warnings.__exit__(None, None, None)\n   \n'
E           A docstring
E         -    
E         -    foo
E         -    
E         -    
E         -    
E         + 
E         + foo
E         + 
E         + 
E         + 
E         -    .. testsetup::
E         ? ---
E         + .. testsetup::
E         -    
E         + 
E         -        >>> import pytest
E         ? ---
E         +     >>> import pytest
E         -        >>> import warnings
E         ? ---
E         +     >>> import warnings
E         -        >>> _suppress_warnings = pytest.deprecated_call()
E         ? ---
E         +     >>> _suppress_warnings = pytest.deprecated_call()
E         -        >>> _ = _suppress_warnings.__enter__()
E         ? ---
E         +     >>> _ = _suppress_warnings.__enter__()
E         -    
E         + 
E         -    Some text
E         ? ---
E         + Some text
E         -    
E         -    
E         + 
E         + 
E         -    .. testcleanup::
E         ? ---
E         + .. testcleanup::
E         -    
E         + 
E         -        >>> warnings.warn("Avoid error if no doctests to run...", DeprecationWarning)
E         ? ---
E         +     >>> warnings.warn("Avoid error if no doctests to run...", DeprecationWarning)
E         -        >>> _ = _suppress_warnings.__exit__(None, None, None)
E         ? ---
E         +     >>> _ = _suppress_warnings.__exit__(None, None, None)
E         -    
E         +

nibabel/tests/test_deprecator.py:95: AssertionError
```

--- a/nibabel/deprecator.py
+++ b/nibabel/deprecator.py
@@ -3,14 +3,25 @@ from __future__ import annotations
 
 import functools
 import re
+import sys
 import typing as ty
 import warnings
+from textwrap import dedent
 
 if ty.TYPE_CHECKING:  # pragma: no cover
     T = ty.TypeVar('T')
     P = ty.ParamSpec('P')
 
 _LEADING_WHITE = re.compile(r'^(\s*)')
+
+
+def _dedent_docstring(docstring):
+    """Compatibility with Python 3.13+.
+
+    xref: https://github.com/python/cpython/issues/81283
+    """
+    return '\n'.join([dedent(line) for line in docstring.split('\n')])
+
 
 TESTSETUP = """
 
@@ -31,6 +42,10 @@ TESTCLEANUP = """
     >>> _ = _suppress_warnings.__exit__(None, None, None)
 
 """
+
+if sys.version_info >= (3, 13):
+    TESTSETUP = _dedent_docstring(TESTSETUP)
+    TESTCLEANUP = _dedent_docstring(TESTCLEANUP)
 
 
 class ExpiredDeprecationError(RuntimeError):

--- a/nibabel/tests/test_deprecator.py
+++ b/nibabel/tests/test_deprecator.py
@@ -14,12 +14,21 @@ from nibabel.deprecator import (
     Deprecator,
     ExpiredDeprecationError,
     _add_dep_doc,
+    _dedent_docstring,
     _ensure_cr,
 )
 
 from ..testing import clear_and_catch_warnings
 
 _OWN_MODULE = sys.modules[__name__]
+
+func_docstring = (
+    f'A docstring\n   \n   foo\n   \n{indent(TESTSETUP, "   ", lambda x: True)}'
+    f'   Some text\n{indent(TESTCLEANUP, "   ", lambda x: True)}'
+)
+
+if sys.version_info >= (3, 13):
+    func_docstring = _dedent_docstring(func_docstring)
 
 
 def test__ensure_cr():
@@ -92,11 +101,7 @@ class TestDeprecatorFunc:
         with pytest.deprecated_call() as w:
             assert func(1, 2) is None
             assert len(w) == 1
-        assert (
-            func.__doc__
-            == f'A docstring\n   \n   foo\n   \n{indent(TESTSETUP, "   ", lambda x: True)}'
-            f'   Some text\n{indent(TESTCLEANUP, "   ", lambda x: True)}'
-        )
+        assert func.__doc__ == func_docstring
 
         # Try some since and until versions
         func = dec('foo', '1.1')(func_no_doc)

